# Customers’ credit worthiness
(preprocessing data)


## Introduction

**Project Description**<br>
Your project is to prepare a report for a bank’s loan division. You’ll need to find out if a customer’s marital status and number of children has an impact on whether they will default on a loan. The bank already has some data on customers’ credit worthiness.
Your report will be considered when building a credit scoring of a potential customer. A credit scoring is used to evaluate the ability of a potential borrower to repay their loan.

**Project Description**

- children : the number of children in the family
- days_employed: how long the customer has worked
- dob_years: the customer’s age
- education: the customer’s education level
- education_id: identifier for the customer’s education
- family_status: the customer’s marital status
- family_status_id: identifier for the customer’s marital status
- gender: the customer’s gender
- income_type: the customer’s income type
- debt: whether the client has ever defaulted on a loan
- total_income: annual income
- purpose: reason for taking out a loan

*Libraries*

In [ ]:
import pandas as pd
import pymystem3 as pm
import collections as col

## General Information of Data.

**Let's check which data we have**

In [1]:
credit_worthiness = pd.read_csv('/datasets/data.csv')
credit_worthiness.to_csv('data.csv', index=False)

In [2]:
credit_worthiness.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


In [3]:
credit_worthiness.duplicated().sum()

54

In [4]:
credit_worthiness.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [5]:
credit_worthiness.head(10)

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0    -926.185831         27    высшее             0   
6         0   -2879.202052         43    высшее             0   
7         0    -152.779569         50   СРЕДНЕЕ             1   
8         2   -6929.865299         35    ВЫСШЕЕ             0   
9         0   -2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   
5  гражданский брак                 1      M   компаньон     0  255763.565419   
6   женат / замужем                 0      F   компаньон     0  240525.971920   
7   женат / замужем                 0      M   сотрудник     0  135823.934197   
8  гражданский брак                 1      F   сотрудник     0   95856.832424   
9   женат / замужем                 0      M   сотрудник     0  144425.938277   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

In [6]:
credit_worthiness.tail(10)

children  days_employed  dob_years       education  education_id  \
21515         1    -467.685130         28         среднее             1   
21516         0    -914.391429         42          высшее             0   
21517         0    -404.679034         42          высшее             0   
21518         0  373995.710838         59         СРЕДНЕЕ             1   
21519         1   -2351.431934         37  ученая степень             4   
21520         1   -4529.316663         43         среднее             1   
21521         0  343937.404131         67         среднее             1   
21522         1   -2113.346888         38         среднее             1   
21523         3   -3112.481705         38         среднее             1   
21524         2   -1984.507589         40         среднее             1   

          family_status  family_status_id gender income_type  debt  \
21515   женат / замужем                 0      F   сотрудник     1   
21516   женат / замужем                 0      F   компаньон     0   
21517  гражданский брак                 1      F   компаньон     0   
21518   женат / замужем                 0      F   пенсионер     0   
21519         в разводе                 3      M   сотрудник     0   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

        total_income                            purpose  
21515  109486.327999              заняться образованием  
21516  322807.776603               покупка своего жилья  
21517  178059.553491       на покупку своего автомобиля  
21518  153864.650328               сделка с автомобилем  
21519  115949.039788  покупка коммерческой недвижимости  
21520  224791.862382                  операции с жильем  
21521  155999.806512               сделка с автомобилем  
21522   89672.561153                       недвижимость  
21523  244093.050500       на покупку своего автомобиля  
21524   82047.418899              на покупку автомобиля

In [7]:
credit_worthiness.describe()

children  days_employed     dob_years  education_id  \
count  21525.000000   19351.000000  21525.000000  21525.000000   
mean       0.538908   63046.497661     43.293380      0.817236   
std        1.381587  140827.311974     12.574584      0.548138   
min       -1.000000  -18388.949901      0.000000      0.000000   
25%        0.000000   -2747.423625     33.000000      1.000000   
50%        0.000000   -1203.369529     42.000000      1.000000   
75%        1.000000    -291.095954     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  1.935100e+04  
mean           0.972544      0.080883  1.674223e+05  
std            1.420324      0.272661  1.029716e+05  
min            0.000000      0.000000  2.066726e+04  
25%            0.000000      0.000000  1.030532e+05  
50%            0.000000      0.000000  1.450179e+05  
75%            1.000000      0.000000  2.034351e+05  
max            4.000000      1.000000  2.265604e+06

According to  the database and our goal we need five columns: "debt", "children", "family_status", "total_income" and "purpose"

**Let's get more information about these columns.**

In [8]:
credit_worthiness['debt'].describe()

count    21525.000000
mean         0.080883
std          0.272661
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max          1.000000
Name: debt, dtype: float64

In [9]:
credit_worthiness['family_status'].describe()

count               21525
unique                  5
top       женат / замужем
freq                12380
Name: family_status, dtype: object

In [10]:
credit_worthiness.sort_values('children')['children'].value_counts().sort_index()

-1        47
 0     14149
 1      4818
 2      2055
 3       330
 4        41
 5         9
 20       76
Name: children, dtype: int64

In [11]:
credit_worthiness.sort_values('family_status')['family_status'].value_counts().sort_index()

Не женат / не замужем     2813
в разводе                 1195
вдовец / вдова             960
гражданский брак          4177
женат / замужем          12380
Name: family_status, dtype: int64

In [12]:
credit_worthiness['total_income'].describe()

count    1.935100e+04
mean     1.674223e+05
std      1.029716e+05
min      2.066726e+04
25%      1.030532e+05
50%      1.450179e+05
75%      2.034351e+05
max      2.265604e+06
Name: total_income, dtype: float64

In [13]:
credit_worthiness[credit_worthiness['total_income'].isnull()]

children  days_employed  dob_years education  education_id  \
12            0            NaN         65   среднее             1   
26            0            NaN         41   среднее             1   
29            0            NaN         63   среднее             1   
41            0            NaN         50   среднее             1   
55            0            NaN         54   среднее             1   
...         ...            ...        ...       ...           ...   
21489         2            NaN         47   Среднее             1   
21495         1            NaN         50   среднее             1   
21497         0            NaN         48    ВЫСШЕЕ             0   
21502         1            NaN         42   среднее             1   
21510         2            NaN         28   среднее             1   

               family_status  family_status_id gender  income_type  debt  \
12          гражданский брак                 1      M    пенсионер     0   
26           женат / замужем                 0      M  госслужащий     0   
29     Не женат / не замужем                 4      F    пенсионер     0   
41           женат / замужем                 0      F  госслужащий     0   
55          гражданский брак                 1      F    пенсионер     1   
...                      ...               ...    ...          ...   ...   
21489        женат / замужем                 0      M    компаньон     0   
21495       гражданский брак                 1      F    сотрудник     0   
21497        женат / замужем                 0      F    компаньон     0   
21502        женат / замужем                 0      F    сотрудник     0   
21510        женат / замужем                 0      F    сотрудник     0   

       total_income                           purpose  
12              NaN                   сыграть свадьбу  
26              NaN                       образование  
29              NaN  строительство жилой недвижимости  
41              NaN  сделка с подержанным автомобилем  
55              NaN                   сыграть свадьбу  
...             ...                               ...  
21489           NaN              сделка с автомобилем  
21495           NaN                           свадьба  
21497           NaN        строительство недвижимости  
21502           NaN  строительство жилой недвижимости  
21510           NaN           приобретение автомобиля  

[2174 rows x 12 columns]

In [14]:
credit_worthiness.sort_values('total_income')['total_income'].value_counts().sort_index().tail()

1.711309e+06    1
1.715018e+06    1
1.726276e+06    1
2.200852e+06    1
2.265604e+06    1
Name: total_income, dtype: int64

In [15]:
credit_worthiness['purpose'].describe()

count       21525
unique         38
top       свадьба
freq          797
Name: purpose, dtype: object

In [16]:
credit_worthiness.sort_values('purpose')['purpose'].value_counts().sort_index()

автомобили                                478
автомобиль                                495
высшее образование                        453
дополнительное образование                462
жилье                                     647
заняться высшим образованием              496
заняться образованием                     412
на покупку автомобиля                     472
на покупку подержанного автомобиля        479
на покупку своего автомобиля              505
на проведение свадьбы                     777
недвижимость                              634
образование                               447
операции с жильем                         653
операции с коммерческой недвижимостью     651
операции с недвижимостью                  676
операции со своей недвижимостью           630
покупка жилой недвижимости                607
покупка жилья                             647
покупка жилья для сдачи                   653
покупка жилья для семьи                   641
покупка коммерческой недвижимости 

*We don't need all data to answer question, but let's check all columns just in case*

In [17]:
credit_worthiness['income_type'].describe()

count         21525
unique            8
top       сотрудник
freq          11119
Name: income_type, dtype: object

In [18]:
credit_worthiness.sort_values('income_type')['income_type'].value_counts().sort_index()

безработный            2
в декрете              1
госслужащий         1459
компаньон           5085
пенсионер           3856
предприниматель        2
сотрудник          11119
студент                1
Name: income_type, dtype: int64

In [19]:
credit_worthiness[credit_worthiness['days_employed'] < 0]['days_employed'].count()

15906

In [20]:
credit_worthiness[credit_worthiness['days_employed']> 0].head()

children  days_employed  dob_years education  education_id  \
4          0  340266.072047         53   среднее             1   
18         0  400281.136913         53   среднее             1   
24         1  338551.952911         57   среднее             1   
25         0  363548.489348         67   среднее             1   
30         1  335581.668515         62   среднее             1   

            family_status  family_status_id gender income_type  debt  \
4        гражданский брак                 1      F   пенсионер     0   
18         вдовец / вдова                 2      F   пенсионер     0   
24  Не женат / не замужем                 4      F   пенсионер     0   
25        женат / замужем                 0      M   пенсионер     0   
30        женат / замужем                 0      F   пенсионер     0   

     total_income                                purpose  
4   158616.077870                        сыграть свадьбу  
18   56823.777243     на покупку подержанного автомобиля  
24  290547.235997  операции с коммерческой недвижимостью  
25   55112.757732                   покупка недвижимости  
30  171456.067993  операции с коммерческой недвижимостью

In [21]:
credit_worthiness['days_employed'].describe()

count     19351.000000
mean      63046.497661
std      140827.311974
min      -18388.949901
25%       -2747.423625
50%       -1203.369529
75%        -291.095954
max      401755.400475
Name: days_employed, dtype: float64

In [22]:
credit_worthiness['dob_years'].describe()

count    21525.000000
mean        43.293380
std         12.574584
min          0.000000
25%         33.000000
50%         42.000000
75%         53.000000
max         75.000000
Name: dob_years, dtype: float64

In [23]:
credit_worthiness.sort_values('dob_years')['dob_years'].value_counts().sort_index()

0     101
19     14
20     51
21    111
22    183
23    254
24    264
25    357
26    408
27    493
28    503
29    545
30    540
31    560
32    510
33    581
34    603
35    617
36    555
37    537
38    598
39    573
40    609
41    607
42    597
43    513
44    547
45    497
46    475
47    480
48    538
49    508
50    514
51    448
52    484
53    459
54    479
55    443
56    487
57    460
58    461
59    444
60    377
61    355
62    352
63    269
64    265
65    194
66    183
67    167
68     99
69     85
70     65
71     58
72     33
73      8
74      6
75      1
Name: dob_years, dtype: int64

In [24]:
credit_worthiness['education'].describe()

count       21525
unique         15
top       среднее
freq        13750
Name: education, dtype: object

In [25]:
credit_worthiness.sort_values('education')['education'].value_counts().sort_index()

ВЫСШЕЕ                   274
Высшее                   268
НАЧАЛЬНОЕ                 17
НЕОКОНЧЕННОЕ ВЫСШЕЕ       29
Начальное                 15
Неоконченное высшее       47
СРЕДНЕЕ                  772
Среднее                  711
УЧЕНАЯ СТЕПЕНЬ             1
Ученая степень             1
высшее                  4718
начальное                250
неоконченное высшее      668
среднее                13750
ученая степень             4
Name: education, dtype: int64

In [26]:
credit_worthiness['gender'].describe()

count     21525
unique        3
top           F
freq      14236
Name: gender, dtype: object

In [27]:
credit_worthiness.sort_values('gender')['gender'].value_counts().sort_index()

F      14236
M       7288
XNA        1
Name: gender, dtype: int64

### Conclusion

  *General information*

   - 21525 entries
   - 54 duplicates: it is database of bank where can be customers with equals parameters(personal experience). Because we don't have any column with unique information (even id number) we can't be sure that this is duplicates(or i don't understand what is duplicate)))))
   - missing data in 2 columns: days_employed and total_income. It is equal. Probably, there is a correlation between them, but we don't have any kind of information that proves out guess.
   
  *Information by columns*

   - column "debt" has no incorrect data;
   - column "children" has incorrect number of children like -1 or 20. (We can or delete such data or replace with median. But I think it is better to delete, because such replacement can, probably, influence on future stat analyze.);
   - column "family_status" has upper_case in "Не женат / не замужем", I don't think that it will be a problem;
   - column "purpose" has a lot of variations of the same loan's purposes. We need to make it easier to read;
   -"total_income" has NaN data. I think it is better to replace it with zero(0).


   - column "days_emploed" has mines input which is can't be real. Also we have data that also is too big to be real;
   - column "dob_years" - function describe()showed us min value as 0. I checked that 101 persone is 0 years old. But we don't need that column;
   - column "education". As with previous column describe() function showed weird values as unique = 15. Also i checked this column and find out that we have different font size. We just need to convert it to one font size using lower().



**Notes**:

To prevent getting incorrect data in the future we need:
- check with developer team if there are problems with collecting data in CRM program
- check with colleges why we have incorrect data.

##  Preprocessing Data

### Missing data

I don't think that if we replace missing data 'total_income' to mean or median that will be correct because it can influence on a decision to give a loan or not. That's why I think that it is better to replace missing data to 0(zero) 

In [28]:
credit_worthiness['total_income'] = credit_worthiness['total_income'].fillna(0)

In [29]:
credit_worthiness['total_income'].isnull().sum()

0

Let's delete incorrect data from 'children' column

In [30]:
credit_worthiness = credit_worthiness[(credit_worthiness['children'] >= 0) & (credit_worthiness['children'] <= 5)]

credit_worthiness.sort_values('children')['children'].value_counts().sort_index()

0    14149
1     4818
2     2055
3      330
4       41
5        9
Name: children, dtype: int64

#### Conclusion

Delete incorrect data from column 'children'. Also we can try to leave all data and just create a category "error" and see the results.
Replace NaN value to 0(zero). 

### Changing Datatypes

Convert float type into integer using astype()

In [31]:
credit_worthiness['total_income'] = credit_worthiness['total_income'].round().astype('int')

print(credit_worthiness['total_income'].head())

0    253876
1    112080
2    145886
3    267629
4    158616
Name: total_income, dtype: int64


#### Conclusion

Converting this data into another type makes it more readable and easy to work.

### Duplicates

In column "education" which is out of our mission we has dublicates. Let's do something with it.

In [32]:
credit_worthiness['education'] = credit_worthiness['education'].str.lower()
credit_worthiness.sort_values('education')['education'].value_counts().sort_index()

высшее                  5237
начальное                282
неоконченное высшее      741
среднее                15136
ученая степень             6
Name: education, dtype: int64

#### Conclusion

All duplicates were deleted automatically after changing font size. Duplicates in "purpose" column will be deleted in next steps.  
**CONCLUSION:** Before delete any data check if there is another options to replace incorrect data to correct!!!!!
As i mentioned earlier I don't think that duplicates in this database is real duplicates. Plus we don't know this database by one office in a city(and which one) or all offices in a city or even in a country. 

### Lemmatization

In [33]:
credit_worthiness['purpose'] = ''.join(pm.Mystem().lemmatize('|'.join(credit_worthiness['purpose']))).split('|')

credit_worthiness['purpose']

0                     покупка жилье
1           приобретение автомобиль
2                     покупка жилье
3        дополнительный образование
4                   сыграть свадьба
                    ...            
21520              операция с жилье
21521           сделка с автомобиль
21522                  недвижимость
21523    на покупка свой автомобиль
21524       на покупка автомобиль\n
Name: purpose, Length: 21402, dtype: object

#### Conclusion

After lemmatizetion we can categorize this data.


### Categorization

*One of the questions: "Is parental status has impact on if customer will default on a loan?". To answer this question first we need to categorize data about parental status. We will have just two categories: "have kids", "no kids".*

In [34]:
def category_child(child):
    if child > 0:
        return "have kids"

    return "no kids"

credit_worthiness['category_child'] = credit_worthiness['children'].apply(category_child)

print(credit_worthiness.head())

   children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253876   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145886   
3   женат / замужем                 0      M   сотрудник     0        267629   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose category_child  
0               покупка жилье      have kids  
1     приобретение автомобиль     

*The same with family status. Here i see two categories: 'couple' and 'single'.*

In [35]:
def category_family(status):
    if status == 'женат / замужем' or status == 'гражданский брак':
        return 'couple'
    else:
        return 'single'

credit_worthiness['category_family'] = credit_worthiness['family_status'].apply(category_family)

print(credit_worthiness.tail(15))

       children  days_employed  dob_years            education  education_id  \
21510         2            NaN         28              среднее             1   
21511         0    -612.569129         29               высшее             0   
21512         0    -165.377752         26               высшее             0   
21513         0   -1166.216789         35              среднее             1   
21514         0    -280.469996         27  неоконченное высшее             2   
21515         1    -467.685130         28              среднее             1   
21516         0    -914.391429         42               высшее             0   
21517         0    -404.679034         42               высшее             0   
21518         0  373995.710838         59              среднее             1   
21519         1   -2351.431934         37       ученая степень             4   
21520         1   -4529.316663         43              среднее             1   
21521         0  343937.404131         6

*Before we can categorize "total_income" data we need to get some information. We know that min = 0, lets find out median, max and just in case avg.*

In [36]:
total_income_median = credit_worthiness['total_income'].median()
credit_worthiness['total_income'].median()

135454.5

In [37]:
total_income_mean = credit_worthiness['total_income'].mean()
print(total_income_mean)

150533.34632277358


In [38]:
total_income_max = credit_worthiness['total_income'].max()
print(total_income_max)

2265604


In [39]:
#credit_worthiness.sort_values('total_income')['total_income'].value_counts().sort_index().head(20)

*I have max value from the first step, but faster to write a code one more time then scroll through all code. Now using median as avg salary we can create four categories: "below avg", "avg salary", "above avg" and "high salary".*

In [40]:
def category_income(total):
    if total < 40000:
        return 'below avg'
    if total <= total_income_median:
        return 'avg salary'
    if total < 250000:
        return 'above avg'
    return 'high salary'

credit_worthiness['category_income'] = credit_worthiness['total_income'].apply(category_income)

print(credit_worthiness.head())

   children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253876   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145886   
3   женат / замужем                 0      M   сотрудник     0        267629   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose category_child category_family category_income  
0               покупка жилье      have kids     

*And finally we need to categorize purpose of getting loan. I see here four categories: "property", "wedding", "car" and "education".*

In [41]:
def category_purpose(purpose):
    if 'жилье' in purpose or 'недвижимость' in purpose:
        return 'property'
    if 'свадьба' in purpose:
        return 'wedding'
    if 'автомобиль' in purpose:
        return 'car'
    return 'education'

credit_worthiness['category_purpose'] = credit_worthiness['purpose'].apply(category_purpose)

print(credit_worthiness.head())

   children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253876   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145886   
3   женат / замужем                 0      M   сотрудник     0        267629   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose category_child category_family category_income  \
0               покупка жилье      have kids    

#### Conclusion

Now we can use categories to group our data and answer to the questions.

## Answer Questions

- Is there a connection between having kids and repaying a loan on time?

In [42]:
children_debt = credit_worthiness.pivot_table(index='category_child',
                                              values='debt')
print(children_debt)

                    debt
category_child          
have kids       0.092238
no kids         0.075129


#### Conclusion

We got that there is no big differences between families with kids and without kids and there repaying loan on time.

- Is there a connection between marital status and repaying a loan on time?

In [43]:
family_debt = credit_worthiness.pivot_table(index='category_family',
                                              values='debt')
print(family_debt)

                     debt
category_family          
couple           0.079699
single           0.085020


#### Conclusion

The same situation as with parent status we have with family status. The difference not so big.

- Is there a connection between income level and repaying a loan on time?

In [44]:
income_debt = credit_worthiness.pivot_table(index='category_income',
                                              values='debt')
print(income_debt)

                     debt
category_income          
above avg        0.085009
avg salary       0.081521
below avg        0.078740
high salary      0.069385


#### Conclusion

With salary we have very interesting situation. Customers with high salary repaying loan on time is lower then customers with avg salary or above avg.

- How do different loan purposes affect timely loan repayment?

In [45]:
purpose_debt = credit_worthiness.pivot_table(index='category_purpose',
                                              values='debt')
print(purpose_debt)

                      debt
category_purpose          
car               0.093284
education         0.092319
property          0.072356
wedding           0.078306


#### Conclusion

And the last question but not the least. How purpose of loan related to repaying on time?. We can divide our results into two groups: First group includes such categories as: "car" and "education" and second: "property" and "wedding". Customers who get loan for the purpose from first group repaying loan on time more often. But it is also not so big difference.

## General Conclusion

My analyze showed that there is no big difference between repaying loan on time and family status or parent status or salary or purpose.
I'm not familiar with Russian bank system and i need a consultation with some one to be sure that my results/conclusions are correct. Plus for more detailed analyze we need some statistical analyse. 